pkg/kubelet/kuberuntime/kuberuntime_manager_test.go

```diff
@@ -476,7 +486,7 @@ func TestKillPod(t *testing.T) {
 
 	err = m.KillPod(pod, runningPod, nil)
 	assert.NoError(t, err)
-	assert.Equal(t, 3, len(fakeRuntime.Containers))
+	assert.Equal(t, 4, len(fakeRuntime.Containers))
```

Comment:
IMHO "kill container after sidecar" must be reflected in the test case to make sure our impl has intended behavior

Reply:
Agreed, should be relatively easy to add in a similar style to TestSyncPodWithSidecarsAndInitContainers

Reply:
Can you elaborate? It's not clear to me how to implement this test